<html><h1>Validation FAILED.</h1><p>Build Time: 6 minutes 41 seconds 889 ms</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/c931799bab2b9906844fee820351cf51c9f9f8ba">c931799bab2b9906844fee820351cf51c9f9f8ba</a></p><pre><code> [sync-dir] Synchronizing /opt/dev/projects/github/liferay-binaries-cache-2020/.gradle/caches/modules-2/files-2.1 into /opt/dev/projects/github/liferay-portal-7.1.x/.gradle/caches/modules-2/files-2.1
 [sync-dir] 0 files synchronized in 490ms
     [echo] Executing Gradle task: baseline
     [exec] Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
     [exec] Using proxy squid.lax.liferay.com:3128.
     [exec] &gt; Task :classes UP-TO-DATE
     [exec] &gt; Task :jar SKIPPED
     [exec] &gt; Task :baseline3 FAILED
     [exec] Gradle build finished at 2019-12-05 16:43:41.567.
     [exec] 
     [exec] 
     [exec] FAILURE: Build failed with an exception.
     [exec] 
     [exec] * What went wrong:
     [exec] Execution failed for task ':baseline3'.
     [exec] &gt; Could not resolve all files for configuration ':detachedConfiguration1'.
     [exec]    &gt; Could not resolve com.liferay.portal:com.liferay.portal.impl:[3.0.0,4.0.0).
     [exec]      Required by:
     [exec]          project :
     [exec]       &gt; Failed to list versions for com.liferay.portal:com.liferay.portal.impl.
     [exec]          &gt; Unable to load Maven meta-data from http://repository-cdn.liferay.com/nexus/content/groups/public/com/liferay/portal/com.liferay.portal.impl/maven-metadata.xml.
     [exec]             &gt; Could not get resource 'http://repository-cdn.liferay.com/nexus/content/groups/public/com/liferay/portal/com.liferay.portal.impl/maven-metadata.xml'.
     [exec]                &gt; Could not GET 'http://repository-cdn.liferay.com/nexus/content/groups/public/com/liferay/portal/com.liferay.portal.impl/maven-metadata.xml'.
     [exec]                   &gt; squid.lax.liferay.com: Name or service not known
     [exec] 
     [exec] * Try:
     [exec] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
     [exec] 
     [exec] * Exception is:
     [exec] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':baseline3'.
     [exec] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:166)
     [exec] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:163)
     [exec] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:191)
     [exec] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:156)</code></pre><h5>For full console, click <a href="https://test-1-3.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/193//consoleText">here</a>.</h5></html>

--- a/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_definition_channels.jsp
+++ b/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_definition_channels.jsp
@@ -37,26 +37,28 @@ long[] commerceChannelIds = cpDefinitionChannelDisplayContext.getCommerceChannel
 			<aui:fieldset>
 				<aui:input checked="<%= cpDefinition.isChannelFilterEnabled() %>" label="enable-filter-channels" name="channelFilterEnabled" type="toggle-switch" value="<%= cpDefinition.isChannelFilterEnabled() %>" />
 
-				<c:choose>
-					<c:when test="<%= commerceChannels.isEmpty() %>">
-						<div class="alert alert-info">
-							<liferay-ui:message key="there-are-no-channels" />
-						</div>
-					</c:when>
-					<c:otherwise>
+				<c:if test="<%= cpDefinition.isChannelFilterEnabled() %>">
+					<c:choose>
+						<c:when test="<%= commerceChannels.isEmpty() %>">
+							<div class="alert alert-info">
+								<liferay-ui:message key="there-are-no-channels" />
+							</div>
+						</c:when>
+						<c:otherwise>
 
-						<%
-						for (CommerceChannel commerceChannel : commerceChannels) {
-						%>
+							<%
+							for (CommerceChannel commerceChannel : commerceChannels) {
+							%>
 
-							<aui:input checked="<%= ArrayUtil.contains(commerceChannelIds, commerceChannel.getCommerceChannelId()) %>" label="<%= commerceChannel.getName() %>" name='<%= "commerceChannelId_" + commerceChannel.getCommerceChannelId() %>' onChange='<%= renderResponse.getNamespace() + "fulfillCommerceChannelIds();" %>' type="checkbox" value="<%= commerceChannel.getCommerceChannelId() %>" />
+								<aui:input checked="<%= ArrayUtil.contains(commerceChannelIds, commerceChannel.getCommerceChannelId()) %>" disabled="<%= !cpDefinition.isChannelFilterEnabled() %>" label="<%= commerceChannel.getName() %>" name='<%= "commerceChannelId_" + commerceChannel.getCommerceChannelId() %>' onChange='<%= renderResponse.getNamespace() + "fulfillCommerceChannelIds();" %>' type="checkbox" value="<%= commerceChannel.getCommerceChannelId() %>" />
 
-						<%
-						}
-						%>
+							<%
+							}
+							%>
 
-					</c:otherwise>
-				</c:choose>
+						</c:otherwise>
+					</c:choose>
+				</c:if>
 
 				<aui:button-row>
 					<aui:button cssClass="btn-lg" type="submit" />

